### ISSUE TITLE
Cannot import types

### ISSUE THREAD

**Issue by atlesque** [inrupt/solid-client-vc-js#1028]
### Search terms you've used

types, exports

### Bug description

Cannot import types on a modern Nuxt project.

The types are exported in `package.json` as:

```json
"types": "dist/index.d.ts",
```

We are importing them like:

```ts
import { VerifiableCredential } from '@inrupt/solid-client-vc';
```

But the import fails with a TypeScript error.

### To Reproduce

1. Create a new Nuxt project
2. Add the package as dependency
3. Try to import the types somewhere

### Expected result

Types can be imported and used.

### Actual result

We get the following TypeScript error:

```
Could not find a declaration file for module '@inrupt/solid-client-vc'. '{workDir}/node_modules/@inrupt/solid-client-vc/dist/index.mjs' implicitly has an 'any' type.
  There are types at '/{workDir}/node_modules/@inrupt/solid-client-vc/dist/index.d.ts', but this result could not be resolved when respecting package.json "exports". The '@inrupt/solid-client-vc' library may need to update its package.json or typings.ts(7016)
```

### Environment

```
$ npx envinfo --system --npmPackages --binaries --npmGlobalPackages --browsers

  System:
    OS: macOS 14.4.1
    ...
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    Yarn: 4.1.1 - ~/.nvm/versions/node/v20.10.0/bin/yarn
    npm: 10.5.0 - ~/.nvm/versions/node/v20.10.0/bin/npm
  Browsers:
    Chrome: 124.0.6367.202
    Safari: 17.4.1
  npmPackages:
    @inrupt/solid-client-vc: ^0.7.2 => 0.7.4 
    ...
    nuxt: ^3.10.1 => 3.11.2 
    ...
    typescript: ^5.3.3 => 5.4.5 
    ...
```

## Additional information

A solution is to define the types within the `exports` key in `package.json`:

```json
{
  ...
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "require": "./dist/index.js",
      "import": "./dist/index.mjs"
    },
    ...
  },
  ...
```

**Comment by NSeydoux** [comment]
Hi @atlesque, thanks for reaching out! You're right, it looks like we are missing an entry in the `exports` entry. Our tooling picks up the types from the `types` entry directly, which has been hiding the issue. I'll be fixing this soon, your suggestion is spot on.

**Comment by NSeydoux** [comment]
The fix to this issue should has been released with v1.0.3. Let us know if the issue persists.

**Comment by atlesque** [comment]
Thanks a lot for the quick fix! The types import works correctly now.

Aha, Moebius and Jodorowsky are fantastic - great taste!  🙌
